### PR TITLE
Alerting Notification Policy: Support org-specific provisioning

### PR DESCRIPTION
**Pull request by Cleymax**
Adds the `org_id` attribute to be consistent with other org-specific resources

### CODE REVIEW

File: internal/resources/grafana/resource_alerting_notification_policy.go [hunk 1]
```diff
@@ -179,42 +184,53 @@ func policySchema(depth uint) *schema.Resource {
 }
 
 func readNotificationPolicy(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	client := OAPIGlobalClient(meta) // TODO: Support org-scoped policies
+	client, orgID, _ := OAPIClientFromExistingOrgResource(meta, data.Id())
 
 	resp, err := client.Provisioning.GetPolicyTree()
-	if err != nil {
-		return diag.FromErr(err)
+	if err, shouldReturn := common.CheckReadError("notification policy", data, err); shouldReturn {
```
Comment (julienduchesne) [inline internal/resources/grafana/resource_alerting_notification_policy.go]: Unneeded for notification policy, I think. It should never return 404

File: internal/resources/grafana/resource_alerting_notification_policy_test.go [hunk 2]
```diff
@@ -139,6 +139,57 @@ func TestAccNotificationPolicy_disableProvenance(t *testing.T) {
...
+			},
+		},
+	})
+
+}
+
+func testAccNotificationPolicyInOrg() string {
+	return fmt.Sprintf(`
+	resource "grafana_organization" "test" {
+		name = "test"
```
Comment (julienduchesne) [inline internal/resources/grafana/resource_alerting_notification_policy_test.go]: Template this name with a random string

File: internal/resources/grafana/resource_alerting_notification_policy_test.go [hunk 3]
```diff
@@ -139,6 +139,57 @@ func TestAccNotificationPolicy_disableProvenance(t *testing.T) {
...
+		),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNotificationPolicyInOrg(),
+				Check: resource.ComposeTestCheckFunc(
+					orgCheckExists.exists("grafana_organization.test", &org),
+					alertingNotificationPolicyCheckExists.exists("grafana_notification_policy.test", &policy),
+					checkResourceIsInOrg("grafana_notification_policy.test", "grafana_organization.test"),
+				),
+			},
```
Comment (julienduchesne) [inline internal/resources/grafana/resource_alerting_notification_policy_test.go:181]: Add another step to check that destroying within the org works. See other resources for example

File: internal/resources/grafana/resource_alerting_notification_policy_test.go [hunk 4]
```diff
@@ -139,6 +139,57 @@ func TestAccNotificationPolicy_disableProvenance(t *testing.T) {
...
+	testutils.CheckOSSTestsEnabled(t, ">=9.1.0")
+
+	var policy models.Route
+	var org models.OrgDetailsDTO
+
+	resource.ParallelTest(t, resource.TestCase{
+		ProviderFactories: testutils.ProviderFactories,
+		CheckDestroy: resource.ComposeTestCheckFunc(
+			orgCheckExists.destroyed(&org, nil),
+			alertingNotificationPolicyCheckExists.destroyed(&policy, &org),
```
Comment (julienduchesne) [inline internal/resources/grafana/resource_alerting_notification_policy_test.go]: Unneeded, the org is destroyed so of course the policy will not be queryable

File: internal/resources/grafana/resource_alerting_notification_policy.go (hunk 1 above)
Comment (Cleymax) [reply internal/resources/grafana/resource_alerting_notification_policy.go]: Yes, there will always be a defined receiver.

**Review by julienduchesne — APPROVED** [review]

**Comment by julienduchesne** [comment]
Thanks!

**Pull request merged**